crates/storage/src/connection/event.rs

```diff
@@ -66,14 +68,51 @@ pub struct PageOfEvents {
 }
 
 impl Transaction<'_> {
+    #[cfg(feature = "aggregate_bloom")]
+    pub(super) fn upsert_block_events_aggregate(
```

Comment:
This is pretty sub-optimal. Probably OK for testing query performance but otherwise we should just come up with a way of producing the aggregate in batches (and handle a missing aggregate filter by falling back on checking individual filters?).